[aoti-fx] Add meta["val"] metadata

angelayi opened the pull request:
Summary: Added a `_set_node_metadata_hook` which automatically adds node.meta["val"] to every new node that gets created under this context.

Test Plan:
` buck2 test //mtia/host_runtime/afg/tests:test_dynamic_shapes_advanced_ops`
https://www.internalfb.com/buck2/866439a2-2ba6-42d1-8e43-508d60456e2e

`buck2 test //mtia/host_runtime/afg/tests:test_dynamic_shapes_basic_ops`
https://www.internalfb.com/intern/testinfra/testrun/11540474149662857

Rollback Plan:

Differential Revision: D80579336




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

blaine-rister reviewed (approved):
Nice PR! The FX codegen side LGTM. This is a big generalization/improvement of the way we're handling metadata. 

I will defer the export side to other reviewers.

angelayi committed:
[aoti-fx] Add meta["val"] metadata (#161019)

Summary:

Added a `_set_node_metadata_hook` which automatically adds node.meta["val"] to every new node that gets created under this context. This is done by running the target operator with the fake inputs on its arguments.

Test Plan:
` buck2 test //mtia/host_runtime/afg/tests:test_dynamic_shapes_advanced_ops`
https://www.internalfb.com/buck2/866439a2-2ba6-42d1-8e43-508d60456e2e

`buck2 test //mtia/host_runtime/afg/tests:test_dynamic_shapes_basic_ops`
https://www.internalfb.com/intern/testinfra/testrun/11540474149662857

Rollback Plan:

Reviewed By: blaine-rister

Differential Revision: D80579336
